lib/ipmi_fru.c

```diff
@@ -4767,6 +4767,46 @@ f_type, uint8_t f_index, char *f_string)
 	}
 	/* Convert index from character to decimal */
 	f_index= f_index - 0x30;
+	
+	if (f_type == 'b' && f_index == 9)		// use index '9' for editing 'board mfg date', by slash.wu@ztsystems.com
+	{
+		const uint64_t secs_from_1970_1996 = 820454400;
```

Comment:
This is already defined in `ipmi_fru.h` in `ipmi_fru2time_t()`.
Make it a macro within that header and move the appropriate code there into a new `ipmi_time2fru(time_t *, void *mfg_date)`.